app/dao/notifications_dao.py

```diff
@@ -569,10 +568,7 @@ def dao_update_notifications_by_reference(references, update_dict):
 
 
 @statsd(namespace="dao")
-def dao_get_notifications_by_to_field(service_id, search_term, notification_type=None, statuses=None):
-    if notification_type is None:
-        notification_type = guess_notification_type(search_term)
-
+def dao_get_notifications_by_to_field(service_id, search_term, notification_type, statuses=None):
```

Comment:
should we change function name to: `dao_get_notifications_by_recipient_or_reference` or something like that to reflect that the function does more now?